src/service/fileService.ts

```diff
@@ -0,0 +1,20 @@
+import * as fs from "fs";
+
+export const fileExists = (filePath: string) => {
+  try {
+    if (fs.existsSync(filePath)) {
+      return true;
+    }
+  } catch (err) {
+    console.error(err);
+  }
+  return false;
+};
```

Comment:
Prefer to use a EAFP approach to read files, better than LBYL, because the operations involved in LBYL are not atomic, and the file system might change between the time you look and the time you leap. This prevents TOCTOU problems, see https://en.wikipedia.org/wiki/Time-of-check_to_time-of-use

Reply:
It should be atomic checking if file exists and returning value from the file? Is this what you mean?

Reply:
Nope, I suggest something like (pseudo):

```
try {
  read the file content
} catch (...) {
  execute the fallback reading
}
```

Reply:
Can you check now?

Reply:
Awesome 🥳